src/rust/inetstack/protocols/tcp/peer.rs

```diff
@@ -303,30 +301,36 @@ impl<const N: usize> TcpPeer<N> {
     ) -> Poll<Result<(QDesc, SocketAddrV4), Fail>> {
         let mut inner: RefMut<Inner<N>> = self.inner.borrow_mut();
 
-        let cb: ControlBlock<N> = match inner.qtable.borrow_mut().get_mut(&qd) {
-            Some(InetQueue::Tcp(queue)) => match queue.get_mut_socket() {
+        let cb: ControlBlock<N> = {
+            let mut qtable: RefMut<IoQueueTable> = inner.qtable.borrow_mut();
+            let queue: &mut TcpQueue<N> = qtable.get_mut::<TcpQueue<N>>(&qd)?;
+            match queue.get_mut_socket() {
                 Socket::Listening(socket) => match socket.poll_accept(ctx) {
                     Poll::Pending => return Poll::Pending,
                     Poll::Ready(result) => match result {
                         Ok(cb) => cb,
                         Err(err) => {
-                            inner.qtable.borrow_mut().free(&new_qd);
+                            inner
+                                .qtable
+                                .borrow_mut()
+                                .free::<TcpQueue<N>>(&new_qd)
+                                .expect("queue should exist");
                             return Poll::Ready(Err(err));
                         },
                     },
                 },
                 _ => return Poll::Ready(Err(Fail::new(libc::EOPNOTSUPP, "socket not listening"))),
-            },
-            _ => return Poll::Ready(Err(Fail::new(libc::EBADF, "invalid queue descriptor"))),
+            }
         };
 
         let established: EstablishedSocket<N> = EstablishedSocket::new(cb, new_qd, inner.dead_socket_tx.clone());
         let local: SocketAddrV4 = established.cb.get_local();
         let remote: SocketAddrV4 = established.cb.get_remote();
-        match inner.qtable.borrow_mut().get_mut(&new_qd) {
-            Some(InetQueue::Tcp(queue)) => queue.set_socket(Socket::Established(established)),
-            _ => panic!("Should have been pre-allocated!"),
-        };
+        {
+            let mut qtable: RefMut<IoQueueTable> = inner.qtable.borrow_mut();
+            let new_queue: &mut TcpQueue<N> = qtable.get_mut(&new_qd).expect("Should have been pre-allocated");
```

Comment:
To follow out guidelines, could you add a comment here to explain why it is safe to call `expect()` here?